test/index.test.js

```diff
@@ -148,6 +148,93 @@ describe('mongodb-collection-sample', function() {
     });
   });
 
+  describe('raw buffer', function() {
+    var db;
+
+    before(function(done) {
+      this.timeout(30000);
+      mongodb.MongoClient.connect('mongodb://localhost:27018/test', { useNewUrlParser: true }, function(err, client) {
+        if (err) {
+          return done(err);
+        }
+        db = client.db('test');
+
+        var docs = _range(0, 150).map(function(i) {
+          return {
+            _id: 'needle_' + i,
+            is_even: i % 2,
+            long: bson.Long.fromString('1234567890'),
+            double: 0.23456,
+            int: 1234
+          };
+        });
+        db.collection('haystack').insertMany(docs, done);
+      });
+    });
+
+    after(function(done) {
+      if (!db) {
+        return done();
+      }
+      db.dropCollection('haystack', done);
+    });
+
+    it('should return raw bson buffer when requested', function(done) {
+      sample(db, 'haystack', { size: 10, raw: true })
+        .pipe(es.through(function(doc) {
+          expect(Buffer.isBuffer(doc)).to.be.true;
+        }, function() {
+          this.emit('end');
+          done();
+        }));
+    });
+  });
+
+  // This test creates 2mil documents and samples 1mil of those to make sure
+  // buffer doesn't overflow. Only creating 2mil documents, since any more
+  // causes v8 to run out memory in heap.
+  // FYI: takes a long time to run.
+  describe('raw buffer over a large set of documents', function() {
```

Comment:
one thing that might be work a try: wrap this whole block in a `if (process.env.TEST_BIG_SAMPLER)` or something so its only run if you tell it to specifically and add details to `README`

Reply:
ohhhhhh yes good idea!